protocol/testutil/app/app.go

```diff
@@ -603,6 +660,135 @@ func (tApp *TestApp) PrepareProposal() abcitypes.ResponsePrepareProposal {
 	})
 }
 
+// launchValidator launches a validator using the `start` command with the specified genesis doc and application
+// options. `shutdownFn` must be invoked to cancel the execution of the app. It will block till the application
+// shuts down.
+func launchValidator(
+	genesis types.GenesisDoc,
+	appOptions map[string]interface{},
+) (homeDir string, a *app.App, shutdownFn func() error, err error) {
+	// Create the validators home directory as a temporary directory and fill it with:
+	//  - config/priv_validator_key.json
+	//  - config/node_key.json
+	//  - config/genesis.json
+	validatorHomeDir := filepath.Join(os.TempDir(), fmt.Sprint(time.Now().UnixNano()))
```

Comment:
perhaps silly question, but: is it possible we'll ever get conflicts here? Would something like a uuid4 work better?

Reply:
Its unlikely since I'm using nanoseconds.